File: data/CodeLlama/readable_format_HPEC/fortran/file196.f90
subroutine kernel_trisolv(n , a, x, c)
implicit none

DATA_TYPE, dimension(n, n) :: a
DATA_TYPE, dimension(n) :: c
DATA_TYPE, dimension(n) :: x
integer :: n
integer :: i, j

!$pragma scop
do i = 1, _PB_N
x(i) = c(i)
do j = 1, i - 1
x(i) = x(i) - (a(j, i) * x(j))
end do
x(i) = x(i) / a(i, i)
end do
!$pragma endscop
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file196.cpp
void kernel_trisolv(int n , double a[][n], double x[n], double c[n])
{
 int i, j;
 for (i = 0; i < n; i++)
 {
 x[i] = c[i];
 for (j = 0; j < i; j++)
 x[i] = x[i] - (a[j][i] * x[j]);
 x[i] = x[i] / a[i][i];
 }
}
